packages/codec/lib/compilations/utils.ts

```diff
@@ -4,10 +4,11 @@ const debug = debugModule("codec:compilations:utils");
 import * as Ast from "@truffle/codec/ast";
 import * as Compiler from "@truffle/codec/compiler";
 import { ContractObject as Artifact } from "@truffle/contract-schema/spec";
+import { CompiledContract } from "@truffle/compile-common";
 import { Compilation, Contract, Source } from "./types";
 
 export function shimArtifacts(
-  artifacts: Artifact[],
+  artifacts: (Artifact | CompiledContract)[],
```

Comment:
Don't we need to shim one way or the other here, for bytecodes? Since artifacts use string bytecodes and `CompiledContract` uses `Bytecode` objects

Reply:
Note: these shims are now in @truffle/compile-common itself, which will have to turn into a regular `dependency`

Reply:
No, because it's OK with outputting either, so there's no need to shim.